jwql/utils/utils.py

```diff
@@ -43,6 +43,109 @@
+                 },
+            "filesystem": {"type": "string"},
+            "preview_image_filesystem": {"type": "string"},
+            "thumbnail_filesystem": {"type": "string"},
+            "outputs": {"type": "string"},
+            "jwql_dir": {"type": "string"},
+            "admin_account": {"type": "string"},
+            "log_dir": {"type": "string"},
+            "test_dir": {"type": "string"},
+            "test_data": {"type": "string"},
+            "setup_file": {"type": "string"},
+            "auth_mast": {"type": "string"},
+            "client_id": {"type": "string"},
+            "client_secret": {"type": "string"},
+            "mast_token": {"type": "string"},
+        },
+        # List which entries are needed (all of them)
+        "required": ["connection_string", "database", "filesystem",
+                     "preview_image_filesystem", "thumbnail_filesystem",
+                     "outputs", "jwql_dir", "admin_account", "log_dir",
+                     "test_dir", "test_data", "setup_file", "auth_mast",
+                     "client_id", "client_secret", "mast_token"]
+    }
+
+    # Test that the provided config file dict matches the schema
+    try:
+        jsonschema.validate(instance=config_file_dict, schema=schema)
+    except jsonschema.ValidationError as e:
+        raise jsonschema.ValidationError(
+            'Provided config.json does not match the ' + \
+            'required JSON schema: {}'.format(e.message)
+        )
+
+
+def get_config():
+    """Return a dictionary that holds the contents of the ``jwql``
+    config file.
+
+    Returns
+    -------
+    settings : dict
+        A dictionary that holds the contents of the config file.
+    """
+    config_file_location = os.path.join(__location__, 'config.json')
+
+    # Make sure the file exists
+    if not os.path.isfile(config_file_location):
+        raise FileNotFoundError('The JWQL package requires a configuration file (config.json) '
+                                'to be placed within the jwql/utils directory. '
+                                'This file is missing. Please read the relevant wiki page '
+                                '(https://github.com/spacetelescope/jwql/wiki/'
+                                'Config-file) for more information.')
+
+    with open(config_file_location, 'r') as config_file_object:
+        try:
+            # Load it with JSON
+            settings = json.load(config_file_object)
+        except json.JSONDecodeError as e:
+            # Raise a more helpful error if there is a formatting problem
+            raise ValueError('Incorrectly formatted config.json file. '
+                             'Please fix JSON formatting: {}'.format(e))
+
+    # Ensure the file has all the needed entries with expected data types
+    _validate_config(settings)
+
+    return settings
+
+if not ON_GITHUB_ACTIONS:
+    FILESYSTEM = get_config()['filesystem']
```

Comment:
Nice way to make this run more smoothly! Minor note- I think pep8 wants two spaces before the if statement.